### DISCUSSION TITLE
Does geolocation need a port?

### DISCUSSION THREAD

**Discussion by kerns** [plausible/analytics#885] [Self-Hosted Support] [answered]
Sorry for the dumb question(s), but I am completely new to Docker and learning by doing, reading, failing and asking.

<img width="1320" alt="Screen Shot 2021-03-27 at 20 10 37" src="https://user-images.githubusercontent.com/20254/112731581-80656000-8f38-11eb-87ef-d5340a4a19f6.png">

Just wondering if the presumed lack of network access here is the reason geolocation isn't working for me.

I also tried to run `sudo docker cp plausible_geoip_1:/geoip/GeoLite2-Country.mmdb .` which returns `Error: No such container:path: plausible_geoip_1:/geoip/GeoLite2-Country.mmdb` ...could be bc the file does not exist or could also be this is not how calls into a file system work.

**Comment by Vigasaurus** [comment]
It doesn't need a connected port, the only real reason I could see (from what you've said) is that you don't have your Maxmind credentials set up correctly, see [documentation](https://plausible.io/docs/self-hosting#1-maxmind-geolocation-database) . 

Past that, maybe try sharing your docker-compose.yml (both; with sensitive data redacted) - or the logs you get from Maxmind when spinning up the container.

**Reply by kerns** [reply]
@Vigasaurus I don't know if this helps... I can confirm my geoip-conf.env is also in order with the necessary credentials.

[docker-compose.yml](https://github.com/plausible/analytics/files/6217091/docker-compose.yml.txt)

[docker-compose.geoip.yml](https://github.com/plausible/analytics/files/6217086/docker-compose.geoip.yml.txt)

I am launching the containers via `sudo docker-compose -f docker-compose.yml -f geoip/docker-compose.geoip.yml up` ...in that process and the long scramble of text that follows I get one of the only references to geoip here, and the creation of a file at a path I don't recognize. Maybe that is relevant. (?)

<img width="1124" alt="Screen Shot 2021-03-28 at 11 39 56" src="https://user-images.githubusercontent.com/20254/112748382-8d776300-8fbb-11eb-9d42-00de315c3107.png">

How and where should I otherwise locate and check logs?

**Reply by kerns** [reply]
How should one go about the task of checking for the successful download / correct location of GeoLite2-Country.mmdb ? Does `sudo docker cp plausible_geoip_1:/geoip/GeoLite2-Country.mmdb .` sound...(sorry)... _plausible_? 😎

**Reply by ukutaht** [reply]
Try this: `docker exec -it plausible_geoip_1 ls /usr/share/GeoIP/`. It should list `GeoLite2-Country.mmdb` as the only entry. If this is correct, it means your GeoIP database has been downloaded successfully.

The next question is whether it's correctly made available to the `plausible_plausible_1` container. To check that you can run: `docker exec -it plausible_plausible_1 ls /geoip` and it should list the same file.

Let me know what the output of these commands is so I can assist further.

**Reply by kerns** [reply]
@ukutaht That works. 🙏 File is there. Everything is also suddenly working. I think it must have needed a full day (or hour?) of data before it could display the [DETAILS] link on the map (?).

**Reply by ukutaht** [reply]
awesome, glad it's working!

**Comment by kerns** [comment, marked as answer]
No. :)
